circuit/src/builder/expression_builder.rs

```diff
@@ -98,6 +98,29 @@ where
         expr_id
     }
 
+    /// Adds a witness hint to the graph.
+    /// It will allocate a `WitnessId` during lowering, with no primitive op.
+    #[allow(unused_variables)]
+    pub fn add_witness_hint(&mut self, label: &'static str) -> ExprId {
+        let expr_id = self.graph.add_expr(Expr::Witness);
+
+        #[cfg(debug_assertions)]
+        self.allocation_log.push(AllocationEntry {
+            expr_id,
+            alloc_type: AllocationType::WitnessHint,
+            label,
```

Comment:
```suggestion
            _label,
```

Reply:
Underscore implies it's never used, this is not necessarily the case